omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -38,8 +38,33 @@ export default class LatestObsForConceptListController {
                 // Don't add obs older than maxAge
                 if (angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays) {
                     // Add last obs for concept to list
-
-                    if (['8d4a505e-c2cc-11de-8d13-0010c6dffd0f',
+                	
+                	if (angular.isDefined(obs.groupMembers) && obs.groupMembers != null) {
+                        // If obs is obs group
+                        obs.members = [];
+                        obs.value = "";
+                        angular.forEach(obs.groupMembers, (member) => {
+                        	let prefix;
+                        	
+                        	// Formatting the obs with concept prefix
+                        	if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "FSN") {
+                        		prefix = "(" + member.concept.display + " #" + obs.value.split(member.concept.display).length + ") ";
+                        	} else if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "shortName") {
+                        		prefix = "(" + member.concept.name.display + " #" + obs.value.split(member.concept.name.display).length + ") ";
+                        	} else {
+                                // for default  or obsGroupLabels = none option
+                                prefix = "";
+                        	}
+                        	
+                            // this will help determine the count of obs with same concept
+                            if (obs.value === "") {
```

Comment:
Ah ok, we don't need this whole logic here. The `#1`, `#2`, `#3`, ... etc were supposed to be integral part of the FSNs or short names, I used those counts to illustrate the case of an obs group made of members defined by separate/different concepts. I didn't mean that they should be counted on the display. That should simplify this whole code quite a bit.